src/test/mgr/test_progress.py

```diff
@@ -0,0 +1,212 @@
+#python unit test
+import unittest
+import os
+import sys
+from mock import Mock
+import pytest
+import json
+os.environ['UNITTEST'] = "1"
+sys.path.insert(0, "../../pybind/mgr")
+from progress import module
+
+class TestPgRecoveryEvent(object):
```

Comment:
To make this first test pass, you can make start_epoch another parameter so you don't need to mock out the osdmap, and then use an instance of PgRecoveryEvent you construct in setup(), like this:
```
diff --git a/src/pybind/mgr/progress/module.py b/src/pybind/mgr/progress/module.py
index 1bfe28f..d0eef38 100644
--- a/src/pybind/mgr/progress/module.py
+++ b/src/pybind/mgr/progress/module.py
@@ -126,7 +126,7 @@ class PgRecoveryEvent(Event):
     Always call update() immediately after construction.
     """
 
-    def __init__(self, message, refs, which_pgs, evacuate_osds):
+    def __init__(self, message, refs, which_pgs, evacuate_osds, start_epoch):
         super(PgRecoveryEvent, self).__init__(message, refs)
 
         self._pgs = which_pgs
@@ -139,7 +139,7 @@ class PgRecoveryEvent(Event):
 
         self._progress = 0.0
 
-        self._start_epoch = _module.get_osdmap().get_epoch() 
+        self._start_epoch = start_epoch
 
         self.id = str(uuid.uuid4())
         self._refresh()
@@ -380,7 +380,8 @@ class Module(MgrModule):
                     "Rebalancing after osd.{0} marked {1}".format(osd_id, marked),
                     refs=[("osd", osd_id)],
                     which_pgs=affected_pgs,
-                    evacuate_osds=[osd_id]
+                    evacuate_osds=[osd_id],
+                    start_epoch=new_map.get_epoch()
                     )
             ev.pg_update(self.get("pg_dump"), self.log)
             self._events[ev.id] = ev
diff --git a/src/test/mgr/test_progress.py b/src/test/mgr/test_progress.py
index 2e13987..30e2bfe 100644
--- a/src/test/mgr/test_progress.py
+++ b/src/test/mgr/test_progress.py
@@ -15,17 +15,9 @@ class TestPgRecoveryEvent(object):
     def setup(self):
         # Creating the class and Mocking 
         # a bunch of attributes for testing
-        self.test_class = module.PgRecoveryEvent
-        self._pgs = [module.PgId(1,i) for i in range(3)]
-        self._evacuate_osds = [0]
-        self._original_pg_count = len(self._pgs)
-        self._original_bytes_recovered = None
-        self.module_class = Mock()
-        self.module_class.info = Mock()
-        self._refresh = Mock()
-        self._progress = 0.0
-        self._message = None
-        self._start_epoch = 30
+        module._module = Mock() # just so Event._refresh() works
+        self.test_event = module.PgRecoveryEvent(None, None, [module.PgId(1,i) for i in range(3)], [0], 30)
+
 
     def test_pg_update(self):
         # Test for a completed event when the pg states show active+clear
@@ -85,8 +77,8 @@ class TestPgRecoveryEvent(object):
         ]
         }
 
-        self.test_class.pg_update(self,pg_dump,self.module_class)
-        assert self._progress == 1.0
+        self.test_event.pg_update(pg_dump, Mock())
+        assert self.test_event._progress == 1.0
 
 class OSDMap: 
     

```